eo-parser/src/main/resources/org/eolang/parser/add-probes.xsl

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+The MIT License (MIT)
+
+Copyright (c) 2016-2022 Objectionary.com
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:eo="https://www.eolang.org" xmlns:xs="http://www.w3.org/2001/XMLSchema" id="add-probes" version="2.0">
+  <!--
+  !Note: This should always followed by remove-dup-probes.xsl
+  For every object which starts with '.' add probe meta
+  with fully qualified name of the object.
+  Example:
+   For object:
+   <o base=".abc" line="23" ... >
+     <o base="Q" ... >
+     </o>
+   </o>
+   Meta will be added:
+   <meta line="23">
+     <head>probe</head>
+     <tail>Q.abc</tail>
+     <part>Q.abc</part>
+   </meta>
+  -->
+  <xsl:output encoding="UTF-8" method="xml"/>
+  <xsl:function name="eo:qualify" as="xs:string">
+    <xsl:param name="e" as="element()"/>
+    <xsl:variable name="fco" select="$e/o[1]"/>
+    <xsl:choose>
+      <xsl:when test="starts-with($e/@base, '.')">
+        <xsl:value-of select="concat(eo:qualify($fco), $e/@base)"/>
+      </xsl:when>
+      <xsl:otherwise>
+        <xsl:value-of select="concat($fco, $e/@base)"/>
+      </xsl:otherwise>
+    </xsl:choose>
+  </xsl:function>
+  <xsl:template match="/program/metas">
+    <xsl:copy>
+      <xsl:apply-templates select="node()|@*"/>
+      <xsl:for-each select="//o[starts-with(@base, '.')]">
+        <xsl:element name="meta">
+          <xsl:attribute name="line">
+            <xsl:value-of select="@line"/>
+          </xsl:attribute>
+          <xsl:element name="head">
+            <xsl:text>probe</xsl:text>
+          </xsl:element>
+          <xsl:element name="tail">
+            <xsl:value-of select="eo:qualify(.)"/>
+          </xsl:element>
+          <xsl:element name="part">
+            <xsl:value-of select="eo:qualify(.)"/>
+          </xsl:element>
+        </xsl:element>
```

Comment:
@mximp you can easily get rid of the `remove-dup-probes.xsl`. Here, calculate the name of the meta first, and put it into `xsl:variable`. Then, do `xsl:if` to check whether such a meta already exists. If not, add a new `<meta`.

Reply:
@yegor256 It will not work exactly like this.. But I've found the solution. Thanks for the hint with variable.